File: SwiftDS/linkedList.swift
//ReileyMeeks
//LinkedList in swift

import Foundation

class Node<T> {
    var value: T
    var next: Node?
    
    init(value: T) {
        self.value = value
    }
}

class LinkedList<T> {
    var head: Node<T>?
    
    // Method to insert a value at the end of the linked list
    func append(value: T) {
        let newNode = Node(value: value)
        
        if head == nil {
            head = newNode
        } else {
            var currentNode = head
            while currentNode?.next != nil {
                currentNode = currentNode?.next
            }
            currentNode?.next = newNode
        }
    }
    
    // Method to print the linked list
    func printList() {
        var currentNode = head
        while currentNode != nil {
            print(currentNode!.value)
            currentNode = currentNode?.next
        }
    }
}

// Testing the linked list
let list = LinkedList<Int>()
list.append(value: 10)
list.append(value: 20)
list.append(value: 30)
list.printList()

File: PythonDS/linkedList.py
#Linked List in Python

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def append(self, data):
        new_node = Node(data)

        if self.head is None:
            self.head = new_node
        else:
            current = self.head
            while current.next is not None:
                current = current.next
            current.next = new_node

    def display(self):
        if self.head is None:
            print("Linked list is empty.")
        else:
            current = self.head
            while current is not None:
                print(current.data, end=" ")
                current = current.next
            print()


# Example usage:
my_list = LinkedList()
my_list.append(5)
my_list.append(10)
my_list.append(15)
my_list.display()
